File: project/Assets/Scripts/Player/PlayerCollision.js
#pragma strict

var gameSetup : GameSetup;

function OnCollisionEnter2D(collision : Collision2D){
	
	for (var contact : ContactPoint2D in collision.contacts) {
		if(contact.collider.tag.Equals('Player')){			
		
			var player1Rendered=gameSetup.player1.GetComponent(SpriteRenderer) as SpriteRenderer;
			if(player1Rendered.sprite.Equals(gameSetup.player1Sprite)){
				player1Rendered.sprite=gameSetup.player2Sprite;
			}else{			
				player1Rendered.sprite=gameSetup.player1Sprite;
			}

			var player2Rendered=gameSetup.player2.GetComponent(SpriteRenderer) as SpriteRenderer;																								
			if(player2Rendered.sprite.Equals(gameSetup.player2Sprite)){
				player2Rendered.sprite=gameSetup.player1Sprite;
			}else{			
				player2Rendered.sprite=gameSetup.player2Sprite;
			}
																											
			break;
		}	
	}	
}

File: project/Assets/Scripts/Player/PlayerCollision.cs
using UnityEngine;
using System.Collections;

public class PlayerCollision : MonoBehaviour {
	
	public GameObject otherPlayer;
	private double lastIntervalPlayer;
	private double updateInterval=0.5;
	// Use this for initialization
	void Start () {	
		lastIntervalPlayer = Time.realtimeSinceStartup;
	}
	
	void OnCollisionEnter2D(Collision2D collision){
		
		if(collision.collider.tag.Equals(otherPlayer.tag)){
			double currentTime=Time.realtimeSinceStartup;
			if(currentTime > lastIntervalPlayer + updateInterval){
				Debug.Log("other player");
				lastIntervalPlayer=currentTime;
			}
		}
	}
	
	// Update is called once per frame
	void Update () {
		
	}
}
